sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/MetricsFilter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
+
+/**
+ * Simple POJO representing a filter for querying metrics.
+ */
+@Experimental(Kind.METRICS)
+@AutoValue
+public abstract class MetricsFilter {
+
+  public Set<String> steps() {
```

Comment:
Hmm... I kind of like that but I also worry about verbosity. I would think that if we had a StepFilter, we would still like to support `addStep(step)` as a short-hand for `addStepFilter(StepFilter.forStep(step))`, in which case we don't need to make that change now, since it would be an implementation change not an API change.